packages/devtools_app_shared/lib/src/service/service_extension_manager.dart

```diff
@@ -325,23 +347,28 @@ final class ServiceExtensionManager with DisposerMixin {
         // of allowed network related exceptions rather than ignoring all
         // exceptions.
       }
+      return true;
     }
 
-    if (isolateRef != _mainIsolate) return;
+    if (isolateRef != _mainIsolate) return false;
 
     final isolate = await _isolateManager.isolateState(isolateRef).isolate;
-    if (isolateRef != _mainIsolate) return;
+    if (isolateRef != _mainIsolate) return false;
 
     // Do not try to restore Dart IO extensions for a paused isolate.
     if (extensions.isDartIoExtension(name) &&
         isolate?.pauseEvent?.kind?.contains('Pause') == true) {
       _callbacksOnIsolateResume.putIfAbsent(isolateRef, () => []).add(restore);
+      return true;
     } else {
-      await restore();
+      return await restore();
     }
   }
 
-  Future<void> _maybeRestoreExtension(String name, Object? value) async {
+  /// Maybe restores the service extension named [name] with [value].
+  ///
+  /// Returns whether the service extension's state is set.
```

Comment:
okay apologies for the churn here, my comment about this returning a bool was before I understood that we are returning whether the isolate is ready, not whether we actually restored the service extension. So I don't think this method should actually return a bool now, since returning false here does not indicate that the isolate isn't ready, it just indicates that the service extension isn't enabled on the device. So then returning `_maybeRestoreExtension` above affects the logic of the `_restoreExtensionFromDeviceIfReady` which should return the state of the isolate's readiness.

So tl;dr, I think revert this to returning void.

Reply:
Good catch; done.